chef/resource_chef_node.go

```diff
@@ -142,15 +154,18 @@ func resourceChefNodeDelete(d *schema.ResourceData, meta interface{}) error {
 		return fmt.Errorf("Error deleting node: %s", err)
 	}
 
-	req, err := client.NewRequest("DELETE", "clients/"+d.Id(), nil)
-	if err != nil {
-		return fmt.Errorf("Request error: %s", err)
-	}
+	_, err = client.Clients.Get(d.Id())
+	if err == nil {
+		// This should check if the client actually exists before it tries to delete it
+		req, err := client.NewRequest("DELETE", "clients/"+d.Id(), nil)
```

Comment:
I know this came from the other repo, but perhaps we could just do `client.Clients.Delete(d.Id())`? I think it's equivalent to the body of this if.